### PR TITLE
Open source Transcend Consent UI

### PR DESCRIPTION
**Pull request by eligrey**
## Related Issues

- Closes https://transcend.height.app/T-8844

## Public Changelog

_[none]_

## Internal Changelog

_[none]_

## Security Implications

_[none]_

## System Availability

_[none]_

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1 +1,12 @@
-export const TEST = {};
+// These SKIP rules are needed to prevent our reference to airgap.js.d.ts from being removed
+/* SKIP ordered-imports */
```
Comment (michaelfarrell76) [inline src/index.ts:2]: can remove this SKIP ordered-imports stuff

File: src/hooks/useConfig.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { ComponentChildren, createContext, h, JSX } from 'preact';
+import { StateUpdater, useContext, useState } from 'preact/hooks';
+
+// main
+import { ConsentManagerConfig } from '@transcend-io/airgap.js-types';
+
+/**
+ * Config context
+ */
+type TConfigContext = {
```
Comment (michaelfarrell76) [inline src/hooks/useConfig.tsx:11]: nit: interface

File: src/hooks/useAirgap.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import { ComponentChildren, createContext, h, JSX } from 'preact';
+import { StateUpdater, useContext, useState } from 'preact/hooks';
+
+// main
+import { AirgapAPI } from '@transcend-io/airgap.js-types';
+
+/**
+ * Config context
+ */
+type TConfigContext = {
```
Comment (michaelfarrell76) [inline src/hooks/useAirgap.tsx:11]: nit: interface

File: src/@types/airgap.js.d.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
+import type {
```
Comment (michaelfarrell76) [inline src/@types/airgap.js.d.ts:1]: there is a src/.DS_STORE file committed, can we delete that?

File: src/components/MenuItem.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
+// external
+import { ComponentChild, Fragment, h, JSX } from 'preact';
+
+// global
+import { useConfig, useEmotion } from '../hooks';
+
+/**
+ * Props that are always on this element
+ */
+type MenuItemBaseProps = {
```
Comment (michaelfarrell76) [inline src/components/MenuItem.tsx:10]: nit: can we make these all interfaces

File: src/utils/getKeys.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
+// main
+import type { StringKeys } from '@transcend-io/type-utils';
```
Comment (michaelfarrell76) [inline src/utils/getKeys.ts:2]: these are all exposed in type-utils, can we just import getStringKeys, and getKeys from there?

File: src/types.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
+// main
+import type { TrackingPurpose, ViewState } from '@transcend-io/airgap.js-types';
+
+/**
+ * Disclosure of a tracking purpose
+ */
+export type Disclosure = {
```
Comment (michaelfarrell76) [inline src/types.ts:7]: nit: interfaces

File: src/translations/af-ZA.json [hunk 8]
```diff
@@ -0,0 +1,88 @@
+{
```
Comment (michaelfarrell76) [inline src/translations/af-ZA.json:1]: i made a task to handle integration of these translations into our `extract_intl` script in main: https://transcend.height.app/T-10388

Comment (michaelfarrell76) [reply src/translations/af-ZA.json:1]: where are these translations being loaded? at a quick glance i dont see us importing `@transcend-io/internationalization` at all

Comment (eligrey) [reply src/translations/af-ZA.json:1]: Good catch. I need to update this to use our new internationalization package

Comment (anotherminh) [reply src/translations/af-ZA.json:1]: @michaelfarrell76 I thought this tech debt? I vaguely remember Ben making a one-off translation thing

Comment (anotherminh) [reply src/translations/af-ZA.json:1]: o nvm i do see us loading this -- https://github.com/transcend-io/consent-manager-ui/blob/eb909ec079a685aca3940ebe690d5a9f666fe6dd/src/translations/index.ts#L2

Comment (michaelfarrell76) [reply src/translations/af-ZA.json:1]: oh cool that looks good! i may have missed that in the package.json

**Comment by eligrey** [comment]
This PR is now ready to merge after approval! The playground will come in a follow-up PR

**Comment by eligrey** [comment]
small release blocker: can we get https://github.com/transcend-io/logger/pull/1 merged before this PR? I was able to publish the package but not merge that PR yet

**Review by anotherminh — DISMISSED** [review]
🥳

**Comment by eligrey** [comment]
@anotherminh I accidentally dismissed your approval. Can I can a re-approval?

**Review by anotherminh — APPROVED** [review]

**Pull request merged**